packages/react-core/src/filters/FilterTypes.js

```diff
@@ -0,0 +1,74 @@
+import { makeIntervalComplete } from '../utils/makeIntervalComplete';
+
+export const FilterTypes = Object.freeze({
+  IN: 'in',
+  BETWEEN: 'between', // [a, b] both are included
+  CLOSED_OPEN: 'closed_open', // [a, b) a is included, b is not
+  TIME: 'time',
+  STRING_SEARCH: 'stringSearch'
+});
+
+export const filterFunctions = {
+  [FilterTypes.IN](filterValues, featureValue) {
+    return filterValues.includes(featureValue);
+  },
+  [FilterTypes.BETWEEN]: between,
+  [FilterTypes.TIME](filterValues, featureValue) {
+    const featureValueAsTimestamp = new Date(featureValue).getTime();
+    if (isFinite(featureValueAsTimestamp)) {
+      return between(filterValues, featureValueAsTimestamp);
+    } else {
+      throw new Error(`Column used to filter by time isn't well formatted.`);
+    }
+  },
+  [FilterTypes.CLOSED_OPEN]: closedOpen,
+  [FilterTypes.STRING_SEARCH]: stringSearch
+};
+
+// FilterTypes.BETWEEN
+function between(filterValues, featureValue) {
+  const checkRange = (range) => {
+    const [lowerBound, upperBound] = range;
+    return featureValue >= lowerBound && featureValue <= upperBound;
+  };
+
+  return makeIntervalComplete(filterValues).some(checkRange);
+}
+
+// FilterTypes.CLOSED_OPEN
+function closedOpen(filterValues, featureValue) {
+  const checkRange = (range) => {
+    const [lowerBound, upperBound] = range;
+    return featureValue >= lowerBound && featureValue < upperBound;
+  };
+
+  return makeIntervalComplete(filterValues).some(checkRange);
+}
+
+// FilterTypes.STRING_SEARCH
+function stringSearch(filterValues, featureValue, params = {}) {
+  const normalizedFeatureValue = normalize(featureValue, params);
+
+  return filterValues.some((filterValue) => {
+    let stringRegExp = escapeRegExp(normalize(filterValue, params));
+
+    if (params.mustStart) stringRegExp = `^${stringRegExp}`;
+    if (params.mustEnd) stringRegExp = `${stringRegExp}$`;
+
+    const regex = new RegExp(stringRegExp, 'g');
+    return !!normalizedFeatureValue.match(regex);
+  });
+}
+
+// Aux
+function escapeRegExp(value) {
+  return value.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+}
+
+function normalize(data, params) {
+  let normalizedData = '' + data;
```

Comment:
minor, i think that String(data) is more proper constructor nowadays

Reply:
Why?

Reply:
I feel it's a matter of taste haha, I feel confortable with that way, and it's wide known way to cast something to string, but yes, `String(data)` is more explicit. I don't care about what to use, tbh. But in this case, I choose `'' + data` because it's more performant.